gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -33,7 +33,15 @@ roleselection.cv = {
 
 	---
 	-- @realm server
-	ttt_max_baseroles_pct = CreateConVar("ttt_max_baseroles_pct", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "Maximum amount of different baseroles based on player amount. ttt_max_baseroles needs to be 0")
+	ttt_max_baseroles_pct = CreateConVar("ttt_max_baseroles_pct", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "Maximum amount of different baseroles based on player amount. ttt_max_baseroles needs to be 0"),
+
+	---
+	-- @realm server
+	ttt2_roles_shuffle_players = CreateConVar("ttt2_roles_shuffle_players", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "If not enabled, players will get priority for role selection based on the order they connected."),
```

Comment:
I wouldn't add a convar for this, just code it as always enabled. As your solution is an improvement you don't need a convar to renable the "bug"